CMakeLists.txt

```diff
@@ -121,6 +121,9 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     endif()
     string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -fno-strict-aliasing ${SSE_FLAGS} ${AVX_FLAGS}")
   endif()
+  if(PCL_SYMBOL_VISIBILITY_HIDDEN)
+    string(APPEND CMAKE_CXX_FLAGS " -fvisibility=hidden -fvisibility-inlines-hidden")
```

Comment:
Consider setting these flags globally. CMake will figure out right options to pass to compilers.
```cmake
set(CMAKE_CXX_VISIBILITY_PRESET ON)
set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
```

Reply:
Thanks for the suggestion, I will take a look at those CMake options when I find the time to continue this pull request.